File: src/main/java/br/com/rfasioli/micrometermetrics/MicrometerMetricsApplication.java
package br.com.rfasioli.micrometermetrics;

import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
public class MicrometerMetricsApplication {

	@Bean
	JvmThreadMetrics threadMetrics() {
		return new JvmThreadMetrics();
	}

	public static void main(String[] args) {
		SpringApplication.run(MicrometerMetricsApplication.class, args);
	}

}

File: src/main/kotlin/br/com/rfasioli/micrometermetrics/MicrometerMetricsApplication.kt
package br.com.rfasioli.micrometermetrics

import io.micrometer.core.instrument.binder.jvm.JvmThreadMetrics
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean

@SpringBootApplication
class MicrometerMetricsApplication {
    @Bean
    fun threadMetrics() = JvmThreadMetrics()
}

fun main(args: Array<String>) {
    runApplication<MicrometerMetricsApplication>(*args)
}